File: _submissions/19UCS159_Tharun/Day3/redundantBraces.java
import java.util.*;

public class redundantBraces {
  public int braces(String A) {
    Stack<Character> stack = new Stack<>();
    for (char ch : A.toCharArray()) {
      if (ch == ')') {
        int operators = 0;
        while (stack.peek() != '(') {
          char c = stack.pop();
          if (c == '+' || c == '-' || c == '*' || c == '/')
            operators++;
        }
        stack.pop();
        if (operators == 0)
          return 1;
      } else {
        stack.push(ch);
      }
    }
    return 0;
  }
}

File: _submissions/18UEI056_Prashun/Day3/Redundant_Braces.cpp
bool isOperator(char x)
{ if(x=='+'||x=='-'||x=='*'||x=='/')
 return true;
 return false;
    
}

int Solution::braces(string A) {
    stack<char>s;
    for(int i=0;i<A.size();i++)
    { if(A[i]=='(')
     s.push('(');
     else if(isOperator(A[i]) && s.size()>0)
      s.pop();
    }
if(s.empty()) return 0;
else return 1;
    
}
